opentelemetry/spring-boot-instrumentation/otel-collector-config.yaml

```diff
@@ -0,0 +1,109 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+receivers:
+  # Receive OTLP from our application
+  otlp:
+    protocols:
+      grpc:
+        endpoint: "0.0.0.0:4317"
+  # Use the filelog receiver to read our log from its log file.
+  filelog:
+    start_at: beginning
+    include:
+    - "/var/log/app.log"
+    operators:
+      - type: json_parser
+        parse_to: body
+        timestamp:
+          parse_from: body.timestamp
+          layout: '%Y-%m-%dT%H:%M:%S.%fZ'
+        severity:
+          parse_from: body.severity
+
+      # set trace_flags to SAMPLED if GCP attribute is set to true
+      - type: add
+        field: body.trace_flags
+        value: "01"
+        if: body["logging.googleapis.com/trace_sampled"] == true
+
+      # parse the trace context fields from GCP attributes
+      - type: regex_parser
+        parse_from: body["logging.googleapis.com/trace"]
+        parse_to: body
+        regex: projects/.+/traces/(?P<trace_id>.*)
+        trace:
+          span_id:
+            parse_from: body["logging.googleapis.com/spanId"]
+
+      # Remove fields that are redundant from translation above
+      - type: remove
+        field: body.timestamp
+      - type: remove
+        field: body.trace_id
+      - type: remove
+        field: body.trace_flags
+      - type: remove
+        field: body.severity
+      - type: remove
+        field: body["logging.googleapis.com/trace"]
+      - type: remove
+        field: body["logging.googleapis.com/spanId"]
+      - type: remove
+        field: body["logging.googleapis.com/trace_sampled"]
+
+exporters:
+  # Export logs and traces using the standard googelcloud exporter
+  googlecloud:
+    project: $GOOGLE_CLOUD_PROJECT
+    # Needed when using Application Default Credentials in Cloud Shell to call Cloud Trace
+    # destination_project_quota: true
+    log:
+      default_log_name: "opentelemetry.io/collector-exported-log"
+  # Export metrics to Google Managed service for Prometheus
+  googlemanagedprometheus:
+    project: $GOOGLE_CLOUD_PROJECT
+
+processors:
+  # Batch telemetry together to more efficiently send to GCP
+  batch:
+    send_batch_max_size: 500
+    send_batch_size: 500
```

Comment:
It's not necessary or recommended for this, the `googlecloud` exporter will handle splitting this into multiple payloads and requests to respect the limits you mentioned if needed.

Reply:
I would highly recommend to place reasonable numbers here. They look random. And given that neither of them is near the allowed limits the claim for "more efficiently" does not held either.
I think that one or more of the other options will serve better as an example:

* add a comment with a link to the OTel batch configuration reference
* adjust the parameters to be closer to allowed limits i.e.
   ```yaml
   batch:
     send_batch_max_size: 10000
     timeout: 1s
   ```
* keep only `timeout` parameter
* update existing comment to explain that batch size can be controlled by time window or size

Reply:
@dashpole I copied these from the Go sample, were these param values chosen for any particular reason? I'm OK to go with the defaults except `timeout` here.

Re adjusting them to be similar to the limits, this is used for all three of metrics, logs, and traces pipelines below. I could make separate processors for each pipeline but I think it's overkill for the sample and will just complicate things.

Reply:
For the purposes of the sample I agree, but a user might copy paste this

Reply:
FYI - The batch size here should actually be optimised for *tracing* over both logging or metrics, as it's the most likely source of overload / need to batch.

I agree with @dashpole it can be removed, but I also think if you wanted to follow any kind of limit/restriction, it would be tracing limits.

Right now, TRACE quota [details](https://cloud.google.com/trace/docs/quotas) is not limited by this batch processor at all.   The set values of 500/1s lead to reasonable developer experience, but of course should be tuned based on actual collector overhead.

More importantly, batch values likely need to diverge by signal.  I agree with @dashpole on removing, but think this should be fine as-is.